### PR TITLE
Lbe-191 Get all Recordings and download class Recordings

### PR DESCRIPTION
**Pull request by akshatakanaje**

### CODE REVIEW

File: src/main/java/com/numpyninja/lms/controller/ClassController.java [hunk 1]
```diff
@@ -150,7 +154,49 @@ public ResponseEntity<List<ClassRecordingDTO>> getClassRecordingByBatchId(@PathV
...
+
+	@GetMapping(path="/classrecordings")
+	@ApiOperation("Get All Recordings")
+	public ResponseEntity<List<ClassRecordingDTO>> getAllRecordings(){
+		return ResponseEntity.ok(classServices.getClassesRecordings());
+    }
+
+	@GetMapping(path="/downloadrecording")
+	public ResponseEntity<InputStreamResource> downloadRecording(@PathVariable Long csId) throws FileNotFoundException {
+		File filePath = new File("/Users/akshatakanaje/Downloads/sqlhackthons.mp4");
```
Comment (gkav01) [inline src/main/java/com/numpyninja/lms/controller/ClassController.java:166]: change file path to find any users home directory (not Akshata's home directory:))- something like this: 
String homeDir = System.getProperty("user.home")
String downLoadedFilename = outputFileName; //Take is as input(DTO parameter)
File filePath = new File(homeDir + "/"+ filename)

**Review by gkav01 — COMMENTED** [review]
Please add Junit cases for this feature.
Please make suggested changes, test and merge code.

File: src/main/java/com/numpyninja/lms/controller/ClassController.java (hunk 1 above)
Comment (akshatakanaje) [reply src/main/java/com/numpyninja/lms/controller/ClassController.java:166]: This API i have added to understand code and i will remove this one

File: src/main/java/com/numpyninja/lms/controller/ClassController.java [hunk 2]
```diff
@@ -150,7 +154,49 @@ public ResponseEntity<List<ClassRecordingDTO>> getClassRecordingByBatchId(@PathV
...
+
+		InputStreamResource resource = new InputStreamResource(new FileInputStream(filePath));
+		HttpHeaders headers = new HttpHeaders();
+		headers.add(HttpHeaders.CONTENT_DISPOSITION, "attachment; filename=" + filePath.getName());
+
+		return ResponseEntity.ok()
+				.headers(headers)
+				.contentLength(filePath.length())
+				.contentType(MediaType.APPLICATION_OCTET_STREAM)
```
[the remark this answers is not in the document]
Comment (akshatakanaje) [reply src/main/java/com/numpyninja/lms/controller/ClassController.java:175]: I tried and its not work.. its downloading as response.bin

File: src/main/java/com/numpyninja/lms/controller/ClassController.java [hunk 3]
```diff
@@ -150,7 +154,49 @@ public ResponseEntity<List<ClassRecordingDTO>> getClassRecordingByBatchId(@PathV
...
+				.body(resource);
+	}
+
+	@GetMapping(path="/download/{csId}")
```
[the remark this answers is not in the document]
Comment (akshatakanaje) [reply src/main/java/com/numpyninja/lms/controller/ClassController.java:179]: yes

**Pull request merged**